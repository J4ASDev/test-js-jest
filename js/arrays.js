const fruits = [ 'manzana', 'banana', 'naranja', 'fresa', 'mango' ]
const colors = [ 'rojo', 'azul', 'verda', 'amarillo', 'morado' , 'marron']

export const arrayFruits = () => fruits;
export const arrayColors = () => colors;
