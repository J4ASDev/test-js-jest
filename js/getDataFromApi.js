import axios from 'axios';

export const urlRickAndMorty = 'https://rickandmortyapi.com/api/character/';
export const apiError = 'https://httpstat.us/';

export const getDataFromApi = url => {
  return axios.get(url)
    .then(({ data }) => {
      return data
    });
};