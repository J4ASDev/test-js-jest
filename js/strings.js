export const text = 'hoy es un gran dia'
