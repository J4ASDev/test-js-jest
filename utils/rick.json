{
  "id": 1,
  "name": "Rick Sanchez",
  "Status": "Alive",
  "species": "Human",
  "gender": "Male"
}