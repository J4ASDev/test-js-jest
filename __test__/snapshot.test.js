import { getCharacter } from '../js/snapshot';
import rick from '../utils/apis';

describe('Snapshots', () => {
  test('Comparando con datos locales', () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });

  test('Tenemos una exepcion dentro del Codigo', () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: "Jonathan Araujo"
    };

    expect(user).toMatchSnapshot({
      id: expect.any(Number)
    });
  });
});

// jest -u: Update data.
