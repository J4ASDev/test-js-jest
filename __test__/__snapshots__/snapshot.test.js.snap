// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots Comparando con datos locales 1`] = `
Object {
  "gender": undefined,
  "id": undefined,
  "name": undefined,
  "species": undefined,
  "status": undefined,
}
`;

exports[`Snapshots Tenemos una exepcion dentro del Codigo 1`] = `
Object {
  "id": Any<Number>,
  "name": "Jonathan Araujo",
}
`;
